pkg/app/app.go

```diff
@@ -72,24 +72,18 @@ func ValidateFlags(bundle string, args []string, cmd *cobra.Command, opt *kobjec
 
 	// Kubernetes specific flags
 	chart := cmd.Flags().Lookup("chart").Changed
-	daemonSet := cmd.Flags().Lookup("daemon-set").Changed
-	replicationController := cmd.Flags().Lookup("replication-controller").Changed
-	deployment := cmd.Flags().Lookup("deployment").Changed
+	// Get the controller
+	controller := cmd.Flags().Lookup("controller").Value.String()
+	log.Debug("Checking validation of controller %s", controller)
 
 	// Check validations against provider flags
 	switch {
 	case provider == "openshift":
 		if chart {
 			log.Fatalf("--chart, -c is a Kubernetes only flag")
 		}
-		if daemonSet {
-			log.Fatalf("--daemon-set is a Kubernetes only flag")
-		}
-		if replicationController {
-			log.Fatalf("--replication-controller is a Kubernetes only flag")
-		}
-		if deployment {
-			log.Fatalf("--deployment, -d is a Kubernetes only flag")
+		if controller == "daemonset" || controller == "replicationcontroller" {
+			log.Fatalf("--controller is a Kubernetes only flag")
```

Comment:
No it's not a Kubernetes only flag :-1: `--controller` is meant to be added for OpenShift as well..